### DISCUSSION TITLE
Display polygon labels in the image itself.

### DISCUSSION THREAD

**Discussion by krishvishal** [wkentaro/labelme#1132] [Ideas / Feature requests]
Right now to know the label assigned to a polygon we have to check by de-selecting that particular instance of the label. This is a trail and error method and takes considerable amount of to recheck which label a particular polygon has been assigned. Is there a way to display label names around each polygon?

**Comment by wkentaro** [comment]
Currently, this feature is not supported.

**Comment by dexception** [comment]
Taking 10x more time to individually click and check category name.

This should be there from the very beginning.

**Comment by ashuezy** [comment]
+1

**Comment by deimsdeutsch** [comment]
+1

**Comment by jamessmith90** [comment]
+1

**Comment by uday60** [comment]
+1

**Comment by suman-19** [comment]
+1

**Comment by yuhorun** [comment]
```               
p.setPen(QColor(168, 34, 3))
p.setFont(QFont('Decorative', 20))
p.drawText(shape.points[0].x(),shape.points[0].y(), shape.label)
```
![image](https://user-images.githubusercontent.com/45552382/65654257-c4795b00-e04a-11e9-9e68-118cc9729e67.png)

if anybody want be helped label image，please tell me，i can help you

**Comment by deimsdeutsch** [comment]
Which file i have to edit for this ?

**Comment by tboenig** [comment]
What @yuhorun suggests sounds good. @yuhorun you could suggest your suggestion @wkentaro in a branch.

**Comment by yuhorun** [comment]
> Which file i have to edit for this ?

unzip, and replace it in  ***\site-packages\labelme\widgets\canvas.py

[canvas.zip](https://github.com/wkentaro/labelme/files/3682269/canvas.zip)

**Comment by sarahmunirj** [comment]
hello can you help me with this, I try to make these changes but they don't appear in the app

**Comment by yuhorun** [comment]
no problem,how can i help you?

**Comment by ratnadeepkedar** [comment]
- Go to Canvas.py file 
- find out the  def paintEvent(self, event):
- add below code in just before for loop and inside loop 
`p.setPen(QtGui.QColor(168, 34, 3))`
`p.setFont(QtGui.QFont('Decorative', 20))`
`Shape.scale = self.scale`
`for shape in self.shapes:`
    `   p.drawText(shape.points[0].x(),shape.points[0].y(), shape.label)`
- You will see the below result 
![image](https://user-images.githubusercontent.com/24845102/121126259-84213700-c845-11eb-8de0-3938c0324e63.png)

**Comment by korhun** [comment]
Thanks, @ratnadeepkedar. I've modified your answer a little:
```
            pixel = 1.0 / self.scale
            p.setFont(QtGui.QFont('Decorative', int(12 * pixel)))
            for shape in self.shapes:
                x, y = (min(shape.points[0].x(), shape.points[1].x()), shape.points[0].y() - pixel * 6) \
                    if shape.shape_type == "rectangle" else (shape.points[0].x(), shape.points[0].y() - pixel * 6)
                color = shape.line_color
                p.setPen(color.darker(200))
                p.drawText(x + pixel * 2, y + pixel * 2, shape.label)
                p.setPen(color)
                p.drawText(x + pixel, y + pixel, shape.label)
                p.setPen(color.lighter(200))
                p.drawText(x, y, shape.label)
```
